src/btree/bt_split.c

```diff
@@ -1389,8 +1389,6 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
     uint64_t recno;
     uint32_t i, slot;
 
-    WT_ASSERT(session, !multi->has_las);
```

Comment:
@quchenhao Do you know why this assertion exists and whether it is legitimate to remove? I got hit by this in my branch too but didn't understand it.

Reply:
My understanding is that when we are in this function, we should be doing update restore eviction. Since we either do update restore or lookaside eviction before durable history, we shouldn't have any content in lookaside. Now we want to change update restore also inserting to lookaside, so this assumption is no longer true.